mysite/announcements/models.py

```diff
@@ -1,24 +1,39 @@
+from django.contrib.auth.models import Group
 from django.core.exceptions import ValidationError
 from django.db import models
-from django.utils.encoding import python_2_unicode_compatible
 from django.utils import timezone
-from django.contrib.auth.models import Group
+from django.utils.encoding import python_2_unicode_compatible
 
 
 @python_2_unicode_compatible
 class Announcements(models.Model):
+
     title = models.CharField(max_length=32, help_text='This will be shown as the title of notification and announcement')
+
     message = models.CharField(max_length=200, help_text='The message you want to convey')
-    date_time_expire = models.DateTimeField('datetime expire', help_text='The date and time you want the announcement to expire')
 
-    added_at = models.DateTimeField(auto_now_add=True)
-    updated_at = models.DateTimeField(auto_now=True)
+    date_time_to_publish = models.DateTimeField('datetime publish', help_text='Schedule date and time for announcement')
+    date_time_expire = models.DateTimeField('datetime expire', help_text='Expiry date and time for announcement')
 
     groups = models.ManyToManyField(Group)
 
+    created_at = models.DateTimeField(editable=False)
+    modified_at = models.DateTimeField(editable=False)
+
+    sent_at = models.DateTimeField(editable=False, null=True, blank=True)
+
+    has_expired = models.BooleanField(default=False, help_text='Marked true when an announcement expires')
+
     def __str__(self):
         return self.title
 
     def clean(self):
-        if self.date_time_expire <= timezone.now():
-            raise ValidationError("The date and time can't be in past")
\ No newline at end of file
+        if (self.date_time_to_publish <= timezone.now()) or (self.date_time_expire <= timezone.now()):
```

Comment:
Should we also have a validation that date_time_to_publish<=date_time_expire?

Reply:
Yes, makes sense